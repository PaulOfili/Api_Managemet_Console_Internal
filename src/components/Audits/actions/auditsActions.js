import types from '../auditConstants';

export const getAuditsLoader = () => {
    return {
        type: types.GET_AUDITS
    }
}

export const getAuditsSuccess = (payload) => {
    return {
        type: types.GET_AUDITS_SUCCESS,
        payload
    }
}

export const getAuditsFailure = (payload) => {
    return function(dispatch){
        dispatch({
            type: types.GET_AUDITS_FAILURE,
             payload
        });
     }   
}

export const setPageSize = (payload) => {
    return {
        type: types.SET_PAGESIZE,
        payload
    }
}

export const setPageNum = (payload) => {
    return {
        type: types.SET_PAGENUM,
        payload
    }
}