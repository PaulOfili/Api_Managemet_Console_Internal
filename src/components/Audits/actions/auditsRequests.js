import request from "../../../request/request";
import api from "../../../request/endpoints";

export const getAudits = (pageNum, pageSize, searchValue) => {
    let options = {}
    if(!searchValue || searchValue === '')
        options = {
            ...api.getAudits(pageNum, pageSize)
        };
    else 
        options = {
            ...api.searchAudits(pageNum, pageSize, searchValue)
        }
    return request.make(options, true)
}