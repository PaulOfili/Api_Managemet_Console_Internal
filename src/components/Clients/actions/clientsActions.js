import types from '../clientsConstants';

export const getClients = () => {
    return {
        type: types.GET_CLIENTS
    }
}

export const getClientsSuccess = (payload) => {
    return function(dispatch){
        dispatch({
             type: types.GET_CLIENTS_SUCCESS,
             payload
        });
     }   
}

export const getClientsFailure = (payload) => {
    return function(dispatch){
       dispatch({
            type: types.GET_CLIENTS_FAILURE,
            payload
       });
    }   
}

export const getClient = () => {
    return {
        type: types.GET_CLIENT
    }
}

export const getClientSuccess = (payload) => {
    return function(dispatch){
        dispatch({
             type: types.GET_CLIENT_SUCCESS,
             payload
        });
     }   
}

export const getClientFailure = (payload) => {
    return function(dispatch){
       dispatch({
            type: types.GET_CLIENT_FAILURE,
            payload
       });
    }   
}


export const getRequested = () => {
    return {
        type: types.GET_REQUESTED
    }
}

export const getRequestedSuccess = (payload) => {
    return {
        type: types.GET_REQUESTED_SUCCESS,
        payload
    }
}


export const getRequestedFailure = (payload) => {
    return {
        type: types.GET_REQUESTED_FAILURE,
        payload
    }
}

export const getApproved = () => {
    return {
        type: types.GET_APPROVED
    }
}

export const getApprovedSuccess = (payload) => {
    return {
        type: types.GET_APPROVED_SUCCESS,
        payload
    }
}


export const getApprovedFailure = (payload) => {
    return {
        type: types.GET_APPROVED_FAILURE,
        payload
    }
}

export const approveResources = () => {
    return {
        type: types.APPROVE_RESOURCES
    }
}

export const approveResourcesSuccess = (payload) => {
    return function(dispatch){
        dispatch({
             type: types.APPROVE_RESOURCES_SUCCESS,
             payload
        });
     }   
}

export const approveResourcesFailure = (payload) => {
    return function(dispatch){
       dispatch({
            type: types.APPROVE_RESOURCES_FAILURE,
            payload
       });
    }   
}

export const declineResources = () => {
    return {
        type: types.DECLINE_RESOURCES
    }
}

export const declineResourcesSuccess = (payload) => {
    return function(dispatch){
        dispatch({
             type: types.DECLINE_RESOURCES_SUCCESS,
             payload
        });
     }   
}

export const declineResourcesFailure = (payload) => {
    return function(dispatch){
       dispatch({
            type: types.DECLINE_RESOURCES_FAILURE,
            payload
       });
    }   
}

export const setClient = (payload) => {
    return function(dispatch){
       dispatch({
            type: types.SET_CLIENT,
            payload
       });
    }   
}