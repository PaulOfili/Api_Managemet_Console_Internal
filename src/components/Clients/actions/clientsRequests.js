import request from "../../../request/request"
import api from "../../../request/endpoints";


export const getProjects = () => { 
    let options = {
        ...api.getProjects(),
    };
    return request.make(options, true)
}

export const getProject = (projectId) => { 
    let options = {
        ...api.getProject(projectId),
    };
    return request.make(options, true)
}

export const getRequested = (projectId) => { 
    let options = {
        ...api.getRequested(projectId),
    };
    return request.make(options, true)
}

export const getApproved = (projectId) => { 
    let options = {
        ...api.getApproved(projectId),
    };
    return request.make(options, true)
}

export const approveResources = (body, projectId) => { 
    let data = body
    let options = {
        ...api.approveResources(projectId),
        data
    };
    return request.make(options, true)
}

export const declineResources = (body, projectId) => { 
    let data = body
    let options = {
        ...api.declineResources(projectId),
        data
    };
    return request.make(options, true)
}
