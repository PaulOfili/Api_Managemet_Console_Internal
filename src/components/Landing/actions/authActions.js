import types from '../constants/account';
import { removeSession } from '../../../session/cookies';

export const login = () => {
    return {
        type: types.ACCOUNT_LOGIN
    };
};
export const accountLogin = (data) => {
    return {
        type: types.ACCOUNT_WELCOME,
        payload: data
    };
};

export const loginFailure = (error) => {
    return {
        type: types.ACCOUNT_LOGIN_FAILURE,
        error
    };
};

export const resolveRole = (payload) => {
    return {
        type: types.RESOLVE_ROLE,
        payload
    };
};

export const logout = () => {
    removeSession()
    return {
        type: types.ACCOUNT_LOGOUT
    }
}

