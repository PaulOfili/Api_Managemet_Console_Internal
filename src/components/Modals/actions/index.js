import types from '../constants';


export const showModal = (type, props) => {
    return {
        type: types.SHOW_MODAL,
        modalType: type,
        modalProps: props
    }
}

export const hideModal = () => {
    return {
        type: types.HIDE_MODAL
    }
}
