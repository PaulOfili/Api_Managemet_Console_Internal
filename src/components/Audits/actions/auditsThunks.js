// import mockprojectApi from '../../../request/mockData';
import * as auditsApi from './auditsRequests';
import * as actions from './auditsActions';
import { showLoading, hideLoading } from 'react-redux-loading-bar';
import { showAlert } from '../../../alert';

export const getAudits = (pageNum, pageSize, searchValue) => {
    return async dispatch => {
            dispatch(actions.getAuditsLoader());
            dispatch(showLoading());
            try {
                const response = await auditsApi.getAudits(pageNum, pageSize, searchValue)
                response && dispatch(actions.getAuditsSuccess(response.data));
                dispatch(hideLoading());
            }
            catch(error){
                dispatch(hideLoading());
                if(error.data && error.data.message){

                    dispatch(showAlert({
                        type:'error',
                        message: error.data.message        
                    }, dispatch))
                }
                else {
                    dispatch(showAlert({
                        type:'error',
                        message: "Unable to get response from server"          
                    }, dispatch))
                }
                dispatch(actions.getAuditsFailure(error));
            }
    }
}