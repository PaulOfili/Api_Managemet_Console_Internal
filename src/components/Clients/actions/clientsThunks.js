// import mockprojectApi from '../../../request/mockData';
import * as projectsApi from './clientsRequests';
import * as actions from './clientsActions';
import { showLoading, hideLoading } from 'react-redux-loading-bar';
import { showAlert } from '../../../alert';

export const getClients = () => {
    return async dispatch => {
            dispatch(actions.getClients());
            dispatch(showLoading());
            try {
                const response = await projectsApi.getProjects();
                response && dispatch(actions.getClientsSuccess(response.data));
                dispatch(hideLoading());
            }
            catch(error){
                dispatch(hideLoading());
                if(error.data && error.data.message){

                    dispatch(showAlert({
                        type:'error',
                        message: error.data.message        
                    }, dispatch))
                }
                else {
                    dispatch(showAlert({
                        type:'error',
                        message: "Unable to get response from server"          
                    }, dispatch))
                }
                dispatch(actions.getClientsFailure(error));
            }
    }
}

export const getClient = (projectId) => {
    return async dispatch => {
            dispatch(actions.getClient(projectId));
            dispatch(showLoading());
            try {
                const response = await projectsApi.getProject(projectId);
                response && dispatch(actions.getClientSuccess(response.data));
                dispatch(hideLoading());
            }
            catch(error){
                dispatch(hideLoading());
                if(error.data && error.data.message){

                    dispatch(showAlert({
                        type:'error',
                        message: error.data.message        
                    }, dispatch))
                }
                else {
                    dispatch(showAlert({
                        type:'error',
                        message: "Unable to get response from server"          
                    }, dispatch))
                }
                dispatch(actions.getClientFailure(error));
            }
    }
}

export const getRequested = (projectId) => {
    return async dispatch => {
            dispatch(actions.getRequested());
            dispatch(showLoading());
            try {
                const response = await projectsApi.getRequested(projectId);
                response && dispatch(actions.getRequestedSuccess(response.data));
                dispatch(hideLoading());
            }
            catch(error){
                dispatch(hideLoading());
                if(error.data && error.data.message){

                    dispatch(showAlert({
                        type:'error',
                        message: `Unable to get response from server. ${error.data.message}.`     
                    }, dispatch))
                }
                else {
                    dispatch(showAlert({
                        type:'error',
                        message: "Unable to get response from server"          
                    }, dispatch))
                }
                dispatch(actions.getRequestedFailure(error));
            }
    }
}

export const getApproved = (projectId) => {
    return async dispatch => {
            dispatch(actions.getApproved());
            dispatch(showLoading());
            try {
                const response = await projectsApi.getApproved(projectId);
                response && dispatch(actions.getApprovedSuccess(response.data));
                dispatch(hideLoading());
            }
            catch(error){
                dispatch(hideLoading());
                if(error.data && error.data.message){

                    dispatch(showAlert({
                        type:'error',
                        message: `Unable to get response from server. ${error.data.message}.`       
                    }, dispatch))
                }
                else {
                    dispatch(showAlert({
                        type:'error',
                        message: "Unable to get response from server"          
                    }, dispatch))
                }
                dispatch(actions.getApprovedFailure(error));
            }
    }
}

export const setResourceList = (newObject) => {
    return dispatch => {
        dispatch(actions.setClient(newObject));
    }
}

export const approveResources = (body, projectId) => {
    return async dispatch => {
            dispatch(actions.approveResources());
            dispatch(showLoading());
            try {
                const response = await projectsApi.approveResources(body, projectId);
                response && dispatch(actions.approveResourcesSuccess(response.data));
                dispatch(getApproved(projectId));
                dispatch(getRequested(projectId));
                dispatch(setResourceList({}))
                dispatch(hideLoading());
            }
            catch(error){
                dispatch(hideLoading());
                if(error.data && error.data.message){

                    dispatch(showAlert({
                        type:'error',
                        message: `${error.data.message}. Note: The project must be live before approving resources.`      
                    }, dispatch))
                }
                else {
                    dispatch(showAlert({
                        type:'error',
                        message: "Unable to get response from server"          
                    }, dispatch))
                }
                dispatch(actions.approveResourcesFailure(error));
            }
    }
}

export const declineResources = (body, projectId) => {
    return async dispatch => {
            dispatch(actions.declineResources());
            dispatch(showLoading());
            try {
                const response = await projectsApi.declineResources(body, projectId);
                response && dispatch(actions.declineResourcesSuccess(response.data));
                dispatch(getApproved(projectId));
                dispatch(getRequested(projectId));
                dispatch(setResourceList({}))
                dispatch(hideLoading());
            }
            catch(error){
                dispatch(hideLoading());
                if(error.data && error.data.message){

                    dispatch(showAlert({
                        type:'error',
                        message: error.data.message        
                    }, dispatch))
                }
                else {
                    dispatch(showAlert({
                        type:'error',
                        message: "Unable to get response from server"          
                    }, dispatch))
                }
                dispatch(actions.declineResourcesFailure(error));
            }
    }
}
