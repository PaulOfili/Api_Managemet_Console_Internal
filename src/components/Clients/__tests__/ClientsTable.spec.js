import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {ClientsTable} from "../ClientsTable";
import * as clientRequests from '../actions/clientsThunks';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    client: {},
    clients: [],
    passportClient: {},
    dispatch: jest.fn(),
    isLoading: false,
    ...clientRequests
  };
  const enzymeWrapper = shallow(<ClientsTable {...props} />);
  return {
    props,
    enzymeWrapper
  };
}

describe("components", () => {
  describe("ClientsTable", () => {
    it("To match snapshot", () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
