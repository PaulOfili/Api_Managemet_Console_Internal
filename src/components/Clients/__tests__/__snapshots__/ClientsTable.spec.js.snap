// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ClientsTable To match snapshot 1`] = `ShallowWrapper {}`;
